prov/efa/src/rdm/rxr_op_entry.c

```diff
@@ -1357,6 +1387,117 @@ int rxr_op_entry_post_remote_read(struct rxr_op_entry *op_entry)
 	return 0;
 }
 
+/**
+ * @brief post RDMA write request(s)
+ *
+ * This function posts write request(s) according to information in op_entry.
+ * Depending on op_entry->bytes_write_total_len and max write size of device,
+ * this function might issue multiple write requests.
+ *
+ * @param[in,out]	op_entry	op_entry that has information of the read request.
+ * 					If write request is successfully submitted,
+ * 					op_entry->bytes_write_submitted will be updated.
+ * @return	On success, return 0
+ * 		On failure, return a negative error code.
+ */
+int rxr_op_entry_post_remote_write(struct rxr_op_entry *op_entry)
+{
+	int err;
+	int iov_idx = 0, rma_iov_idx = 0;
+	size_t iov_offset = 0, rma_iov_offset = 0;
+	size_t write_once_len, max_write_once_len;
+	struct rxr_ep *ep;
+	struct rxr_pkt_entry *pkt_entry;
+
+	assert(op_entry->iov_count > 0);
+	assert(op_entry->rma_iov_count > 0);
+	rxr_op_entry_prepare_to_post_write(op_entry);
```

Comment:
`rxr_op_entry_prepare_to_post_write` cannot be called here.
this function can be called repeated when write encounter -FI_EAGAIN.

`rxr_op_entry_prepare_to_post_write` will reset the counter `bytes_write_submitted` and `bytes_write_completed`.

Reply:
As discussed offline, we will make `rxr_rma_post_shm_write` and `rxr_op_entry_post_remote_write` both as being capable of being called again after returning -FI_AGAIN, but both of these functions are currently not called from a routine like `rxr_op_entry_post_remote_read_or_queue`, and so any failures filter all the way back to the user of libfabric.

See my latest push